### PR TITLE
Refactor ffmpegtools to reduce dependency linkage

### PR DESCRIPTION
**Pull request by stakita**
This PR refactors some of the classes to reduce direct dependencies between them. I'll summarize the changes below.
Note, these changes are structural and should be largely invisible. The only exception to this is the information string (previously `stream`) when using `getMetadataTrack` with json.

### `config.py`

Relocate ffmpeg version logic from `config.py` to `ffmpegtools.py`. This file just queries the config file if it exists and stores the ffprobe and ffmpeg paths if available. This removes ffmpeg specific operations from the concerns of `config.py`.

### `ffmpegtools.py`

Remove the dependency and internal knowledge of the `Config` class from the `FFMpegTools` class. This class now conditionally overrides the ffprobe and ffmpeg paths if provided, otherwise with initialize itself with sensible defaults (based on the plaform). Additionally, this allows for the determination of whether to use the ffprobe json interface to be internal to the `FFMpegTools` class. This allows for the `getMetadataTrackFromJSON` and `getMetadataTrack` methods to be removed and combined into a single method `getMetadataTrack` which uses the json interface as needed. This simplifies the use of the class in the `gpmf.py` classes.

### `gopro2gpx.py`

The `FFMpegTools` class is instantiated at the top-level `main()` function so that it can be injected into the `Parser` class rather than have it be instantiated inline. This should make the `Parser` class easier to test as we can mock the `FFMpegTools` class.

### `gpmf.py`

This now takes the injected `FFMpegTools` dependency. There is now only a unified method `getMetadataTrack` to call so the conditional has been removed.

### CODE REVIEW

File: gopro2gpx/config.py [hunk 1]
```diff
@@ -13,68 +13,38 @@
...
-        self.ffmpeg_cmd = ffmpeg
-        self.ffprobe_cmd = ffprobe
-        # get ffmpeg version
-        self.version = E()
-        self.version.major = 4
-        self.version.medium = 3
-        self.version.minor = 1
-        self.getFfmpegVersion()
-        self.use_json_format = False
-        if self.version.medium >= 4:
```
Comment (stakita) [inline gopro2gpx/config.py:31]: I'm guessing this (`medium`) was a bug. I've interpreted the logic to be all **major** versions of ffprobe above 4 supported json output. Let me know if this assumption is incorrect.

Comment (juanmcasillas) [reply gopro2gpx/config.py:31]: You are right, medium should be 4 instead of 3.

Comment (stakita) [reply gopro2gpx/config.py:31]: What I mean is that the conditional checks the `self.version.medium` field:
```
if self.version.medium >= 4:
```
That is the, middle field of the semver spec. I'm guessing this should be querying the `self.version.major` field and updated the logic as such in `ffmpegtools.py` (I'll add a note below).

File: gopro2gpx/ffmpegtools.py [hunk 2]
```diff
@@ -9,11 +9,44 @@
...
+    def __init__(self, ffprobe=None, ffmpeg=None):
+        default_ffmpeg, default_ffprobe = default_fftools()
+
+        self.ffmpeg = ffmpeg if ffmpeg else default_ffmpeg
+        self.ffprobe = ffprobe if ffprobe else default_ffprobe
+
+        self.version = self.getVersion()
+
+        self.use_json_format = False
+        if self.version.major >= 4:
```
Comment (stakita) [inline gopro2gpx/ffmpegtools.py:37]: Now using the `major` field of the semver comparison.

Comment (juanmcasillas) [reply gopro2gpx/ffmpegtools.py:37]: roger

**Pull request merged**